components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -0,0 +1,665 @@
+      if (!nft_mint_address_) {
+        NftOwnerDone(absl::nullopt);
+        ScheduleWorkOnTask();
+        return;
+      }
+    }
+
+    // Check if domain is minted and its supply is 1.
+    if (!nft_mint_supply_check_done_) {
+      FetchNftSplMint();
+      return;
+    }
+
+    // Find token account for that mint and extract owner from its data.
+    FetchNftTokenOwner();
+    return;
+  }
+
+  // No nft. Find owner of domain and contents of domain's SOL record.
+  if (!domain_name_registry_state_) {
+    FetchDomainRegistryState();
+    return;
+  }
+
+  if (!sol_record_name_registry_state_) {
+    FetchSolRecordRegistryState();
+    return;
+  }
+}
+
+void SnsResolverTask::SetError(SnsResolverTaskError error) {
+  task_error_.emplace(std::move(error));
+}
+
+void SnsResolverTask::NftOwnerDone(absl::optional<SolanaAddress> nft_owner) {
+  nft_owner_check_done_ = true;
+  if (nft_owner) {
+    task_result_.emplace(*nft_owner);
+  }
+}
+
+void SnsResolverTask::FetchNftSplMint() {
+  DCHECK(domain_address_);
+  DCHECK(!nft_owner_check_done_);
+  DCHECK(!nft_mint_supply_check_done_);
+
+  auto internal_callback = base::BindOnce(&SnsResolverTask::OnFetchNftSplMint,
+                                          weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(solana::getAccountInfo(nft_mint_address_->ToBase58()),
+                  std::move(internal_callback),
+                  solana::ConverterForGetAccountInfo());
+}
+
+void SnsResolverTask::OnFetchNftSplMint(APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(api_request_result.body(), &account_info)) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  if (!account_info) {
+    NftOwnerDone(absl::nullopt);
+    return;
+  }
+
+  auto nft_mint = FromBase64<SplMintData>(account_info->data);
+  if (!nft_mint || nft_mint->supply != 1) {
+    NftOwnerDone(absl::nullopt);
+    return;
+  }
+
+  nft_mint_supply_check_done_ = true;
+}
+
+void SnsResolverTask::FetchNftTokenOwner() {
+  DCHECK(nft_mint_address_);
+
+  auto internal_callback = base::BindOnce(
+      &SnsResolverTask::OnFetchNftTokenOwner, weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(getProgramAccounts(*nft_mint_address_),
+                  std::move(internal_callback),
+                  solana::ConverterForGetProrgamAccounts());
+}
+
+void SnsResolverTask::OnFetchNftTokenOwner(
+    APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  auto [parsing_ok, token_owner] =
+      GetTokenOwnerFromGetProgramAccountsResult(api_request_result.body());
+  if (!parsing_ok) {
+    SetError(MakeInternalError());
```

Comment:
Should it be `SetError(ParseErrorResult(api_request_result.body()));` instead?

Reply:
https://github.com/brave/brave-core/pull/15617/commits/1e34bd9363c2e1aa4593c00f3409b8c119ccab29